File: app/src/main/java/com/asmat/rolando/popularmovies/database/WatchLaterMovie.java
package com.asmat.rolando.popularmovies.database;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;
import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(tableName = "watch_later_movies",
        foreignKeys = @ForeignKey(
                entity = Movie.class,
                parentColumns = "id",
                childColumns = "id",
                onDelete = CASCADE))

public class WatchLaterMovie {
    @PrimaryKey
    int id;

    public WatchLaterMovie(int id) {
        this.id = id;
    }
}

File: app/src/main/java/com/asmat/rolando/popularmovies/database/entities/WatchLaterMovie.kt
package com.asmat.rolando.popularmovies.database.entities

import android.arch.persistence.room.*

@Entity(tableName = "watch_later_movies")
data class WatchLaterMovie(val posterPath: String?,
                           val overview: String,
                           val releaseDate: String,
                           @PrimaryKey val id: Int,
                           val title: String,
                           val backdropPath: String?,
                           val popularity: Double)